### PR TITLE
Plot the same projections for genes as the weighted projections

### PR DESCRIPTION
**Pull request by andrewram4287**
Making sure we are plotting (projections * B) for UMAP

### CODE REVIEW

**Comment by andrewram4287** [comment]
@aarmey Were we getting this dependency error before?

**Comment by aarmey** [comment]
Run `poetry update` again and see if it goes away.

**Comment by aarmey** [comment]
You can ignore it for now. Is this PR otherwise good?

**Comment by andrewram4287** [comment]
> You can ignore it for now. Is this PR otherwise good?

Yes, it should be!

File: sccp/figures/figure3a.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def makeFigure():
     CompW = allP @ factors[1]
 
     # UMAP dimension reduction
-    ump = umap.UMAP(random_state=1).fit_transform(allP)
+    ump = umap.UMAP(random_state=1).fit_transform(CompW)
```
Comment (aarmey) [inline sccp/figures/figure3a.py:30]: The UMAP positions should be based only on the projections, not B * projections which are the factor effects.

File: sccp/figures/figure3.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def makeFigure():
     
     # UMAP dimension reduction
     umapReduc = umap.UMAP(random_state=1)
-    pf2Points = umapReduc.fit_transform(np.concatenate(projs, axis=0))
+    pf2Points = umapReduc.fit_transform(np.concatenate(projs, axis=0) @ factors[1])
```
Comment (aarmey) [inline sccp/figures/figure3.py:40]: The UMAP positions should be based only on the projections, not B * projections which are the factor effects.

File: sccp/figures/figure3.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def makeFigure():
     plotDrugUMAP(drugs, "Pf2", dataDF["Drug"].values, pf2Points, ax[8:10])
     plotDrugUMAP(drugs, "PCA", dataDF["Drug"].values, pcaPoints, ax[10:12])
 
-    cmp = ["Cmp. 23", "Cmp. 25", "Cmp. 29", "Cmp. 30"]
+    cmp = ["Cmp. 24", "Cmp. 25", "Cmp. 12", "Cmp. 15"]
```
Comment (aarmey) [inline sccp/figures/figure3.py:60]: Maybe we should just remove these, since the component-by-component cell weights are all plotted in Fig. 3a?

**Review by aarmey — CHANGES_REQUESTED** [review]

**Review by aarmey — APPROVED** [review]

**Pull request merged**